fix: [ADLN] Remove undefined pins from GPIO configuration

bachaue1 opened the pull request:
The ADL-N configuration refers to several pins that (according to
RDC document #648094: "GPIO Implementation Summary") do not exist.
Remove the offending pins both from the GPIO template .yaml file
and the board configuration .dlt files referring to it.

bejeanmo commented:
@bachaue1 is this still an active WIP?

bachaue1 commented:
> @bachaue1 is this still an active WIP?

Yes, this is still relevant (the ADL-N hardware didn't chance since then).

bejeanmo commented:
> > @bachaue1 is this still an active WIP?
> 
> Yes, this is still relevant (the ADL-N hardware didn't chance since then).

If this PR is ready for review, please remove WIP from the title for clarity.

bachaue1 commented:
> > > @bachaue1 is this still an active WIP?
> > 
> > 
> > Yes, this is still relevant (the ADL-N hardware didn't chance since then).
> 
> If this PR is ready for review, please remove WIP from the title for clarity.

Updated title, description (and rebased change to tip of master).
